inst/dev/design_s3.md

```diff
@@ -0,0 +1,35 @@
+# S3 return types for model types
+
+All three models to return an object (`epinowfit` or the like) inheriting from list and containing
+
+- `fit` (the stan object)
```

Comment:
do we want some interim representation between the input args and what is passed to stan? the only use case I can see for that would be printing and for something like an update function

Reply:
Maybe. We've had the problem elsewhere that once we've gone dist_spec -> stan we can't go back so might need to either be able to do that or else keep the original arguments.

We could try without and see how it shakes out / if keeping R level inputs beyond the observation data frames will be needed.